kmk/modules/pimoroni_trackball.py

```diff
@@ -48,76 +48,158 @@
 ANGLE_OFFSET = 0
 
 
+class TrackballHandlerKeyMeta:
+    def __init__(self, handler=0):
+        self.handler = handler
+
+
+def layer_key_validator(handler):
+    return TrackballHandlerKeyMeta(handler=handler)
+
+
 class TrackballMode:
-    '''Behaviour mode of trackball: mouse movement or vertical scroll'''
+    """Behaviour mode of trackball: mouse movement or vertical scroll"""
 
     MOUSE_MODE = const(0)
     SCROLL_MODE = const(1)
 
 
-class Trackball(Module):
-    '''Module handles usage of Trackball Breakout by Pimoroni'''
+class TrackballHandler:
+    def handle(self, keyboard, trackball, x, y, switch, state):
+        raise NotImplementedError
+
+
+class PointingHandler(TrackballHandler):
+    def handle(self, keyboard, trackball, x, y, switch, state):
+        if trackball.mode == TrackballMode.MOUSE_MODE:
+            if x >= 0:
+                trackball.pointing_device.report_x[0] = x
+            else:
+                trackball.pointing_device.report_x[0] = 0xFF & x
+            if y >= 0:
+                trackball.pointing_device.report_y[0] = y
+            else:
+                trackball.pointing_device.report_y[0] = 0xFF & y
+            trackball.pointing_device.hid_pending = x != 0 or y != 0
+        else:  # SCROLL_MODE
+            trackball.pointing_device.report_w[0] = y
+            trackball.pointing_device.hid_pending = y != 0
+
+        if switch == 1:  # Button pressed
+            trackball.pointing_device.button_status[
+                0
+            ] |= trackball.pointing_device.MB_LMB
```

Comment:
```suggestion
            trackball.pointing_device.button_status[0] |= trackball.pointing_device.MB_LMB
```

Reply:
Sorry, but that is a terrible style and the linter should catch that (no offense).

Reply:
none taken, seems that was black trying to adhere to the 80 char per line limit